src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -185,20 +188,38 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                         }
 
                         _currentConfiguration = configuration;
+                    }
+                    catch (IOException ex)
+                    {
+                        _fetchMetadataFailure = ex;
 
+                        // HttpDocumentRetriever throws an IOException on non-successful status code including HttpStatusCode.RequestTimeout and HttpStatusCode.ServiceUnavailable
+                        if (_currentConfiguration == null)
+                        {
+                            if (_bootStrapRetryCount++ < BootstrapRetryCount)
+                            {
+                                var statusCode = ex.Data[HttpDocumentRetriever.StatusCode];
+                                if (statusCode != null && (statusCode.Equals(HttpStatusCode.RequestTimeout) || statusCode.Equals(HttpStatusCode.ServiceUnavailable)))
```

Comment:
>HttpStatusCode.RequestTimeout [](http://example.com/codeflow?start=77&length=29)

I didn't realize this was the HTTP Code Request Timeout 408. I don't think we should special case that code.

The intent of the change should be if there's a network issue suspected, we should retry aggressively for a bit, otherwise let's just use the default.

I think this is adding additional complexity and not actually addressing the root thing we're targeting.

I would think, EITHER:
-Let's just have this be the behavior regardless of the exception type (retry for 2 min @30sec each)
-Let's just wait on this PR and find a better way for the exception handling to only trigger for network issues.

This shouldn't in principle block LKG, they're unrelated.

Reply:
One might argue that we might just try for 2 mins, 30 sec apart for any reason.

Reply:
I think the algorithm could be the automaticRefresh is 30 sec until 4 attempts OR a configuration (valid or invalid) is obtained.

Reply:
+1 for having better way of recognising network issues or have 2min retrying at bootstrap.